IRSA-7299: Update upload id for online help

kpuriIpac opened the pull request:
**Ticket**: https://jira.ipac.caltech.edu/browse/IRSA-7299
**IFE PR**: https://github.com/IPAC-SW/irsa-ife/pull/445
- Updated the upload id from `catalogs.owncatalogs` to `upload.intro`
- This applies to the upload tab as well, and not just the upload dialogs 
  - @lrebull if there's a place where the help link should instead go to `catalogs.owncatalogs` when you test the apps below, please let me know
- **_Note_**: I suppose for [Firefly](https://fireflydev.ipac.caltech.edu/irsa-7299-help-fix-upload-dialog/firefly) we don't have an `upload.intro` for the online help, so maybe I should switch back to the default `basics.searching` for that? 

- Fixes [FIREFLY-1826](https://jira.ipac.caltech.edu/browse/FIREFLY-1826)
  - @loitly the bug fix for this is in DataType's `formatHeader` 
     - Bug summary: Essentially, if you have a spectrum table like this (from the ticket)
        - <img width="722" height="573" alt="photometry_tbl" src="https://github.com/user-attachments/assets/c676f43d-c3aa-4ba0-bd0b-b481fbab540b" />
     -  notice the `det_id` column. When saving this table as an IPAC table, we convert `short` to `integer`, that part works as expected. But when we call `fitValueInto` inside `formatHeader`, the width for this column is returned as 6 (because of the `det_id` string length), and `integer` is truncated into `intege`, hence the bug.
     - This fix takes care of that, while still being generic. 
        - When the `width` is longer than than the `val length` (as is the usual case), for example, for the `ra` column (not in the screenshot), the width is 11 due to the entries in the column, so we still get a nicely formated `"ra         "` as expected in that case. 
       - But in a case like this specific column, it will make sure the data type value is not truncated. 

**Testing**: 
- [Firefly](https://fireflydev.ipac.caltech.edu/irsa-7299-help-fix-upload-dialog/firefly), [Irsaviewer](https://irsa-7299-help-fix-upload-dialog.irsakubedev.ipac.caltech.edu/irsaviewer), [DCE](https://irsa-7299-help-fix-upload-dialog.irsakubedev.ipac.caltech.edu/irsaviewer/dce), [Euclid](https://irsa-7299-help-fix-upload-dialog.irsakubedev.ipac.caltech.edu/applications/euclid), [Spherex](https://irsa-7299-help-fix-upload-dialog.irsakubedev.ipac.caltech.edu/applications/spherex)
- Just test the help link anywhere upload is used takes you to `/onlinehelp/#id=upload.intro`
  - includes both upload tabs and dialogs   
- For [FIREFLY-1826](https://jira.ipac.caltech.edu/browse/FIREFLY-1826), test according to ticket. Saving the table as an IPAC table should fix the bug in the `det_id` column. And if you try to re-upload this column to firefly, that should work as well. 
  - if Spherex searches take too long, I attached a euclid spectrum table to the ticket. I changed the `NDITH` column to be datatype `short` (instead of `integer`). So if you upload this file to firefly, and try to save the table as an IPAC table, you can test this bug fix that way (and to cross verify, if you upload it on the nightly firefly build: https://fireflydev.ipac.caltech.edu/firefly you'll  encounter the bug again)

kpuriIpac committed:
IRSA-7299: Change upload help id and fix ipactbl datatype bug

lrebull commented:
Help anchor: this is definitely fixed for euclid and spherex, which were the ones that were causing the "trouble" with testers. Irsa viewer and dce also work great. so, i need to add the upload.intro anchor to firefly generic help -- that's definitely on me. it should NOT go to basics. 

"intege" type: I confirm this is fixed.

thank you!

kpuriIpac commented:
> Help anchor: this is definitely fixed for euclid and spherex, which were the ones that were causing the "trouble" with testers. Irsa viewer and dce also work great. so, i need to add the upload.intro anchor to firefly generic help -- that's definitely on me. it should NOT go to basics.
> 
> "intege" type: I confirm this is fixed.
> 
> thank you!

Ok so I will leave that as `upload.intro` for firefly as well, once that anchor is created it'll show up.

lrebull commented:
IN case anyone cares (probably not) ... I went through several minutes of being extra confused . . . i got into the firefly docs, and there sure as heck IS already an upload.intro anchor there.
<img width="508" height="470" alt="Screenshot 2025-10-22 at 10 21 26 AM" src="https://github.com/user-attachments/assets/07cd1550-46d8-4569-a636-437f502ec7f4" />

and there is really an upload.html#intro destination for it too. BUT apparently I never actually put the toc_upload section into the fireflyToc. sigh. 

fixed now and checked back into git

loitly commented on src/firefly/java/edu/caltech/ipac/table/DataType.java:
```diff
@@ -446,7 +446,7 @@ public String formatFixedWidth(Object value) {
     public String formatHeader(String val) {
         val = val == null ? "" : val;
-        return fitValueInto(val, getWidth(), false);
+        return fitValueInto(val, Math.max(getWidth(), val.length()), false);
```
This fix doesn’t fully address the issue. The problem occurs where the headers are modified after shrinkToFitData is called.  [here](https://github.com/Caltech-IPAC/firefly/blob/fb9ea7052271175f9fd8f19336ce6581df6210d9/src/firefly/java/edu/caltech/ipac/table/io/IpacTableWriter.java#L67)

I suggest moving that function call to just before the writing step. You may also need to add a second parameter to the function so you can pass in the updated headers for the calculation.

loitly reviewed (approved):
Please fix the bad IPAC table export bug as I suggested before merging.

kpuriIpac commented:
> Please fix the bad IPAC table export bug as I suggested before merging.

@loitly I addressed the feedback and updated the test builds. Can you have a quick look before I merge?

loitly reviewed (approved):
IPAC table changes look good.  Thanks for the fix.
